reconcile/utils/terrascript_aws_client.py

```diff
@@ -1258,10 +1254,42 @@ def populate_tf_resource_rds(self, spec):
             pg_values["parameter"] = pg_values.pop("parameters")
             if self._multiregion_account(account) and len(provider) > 0:
                 pg_values["provider"] = provider
-            pg_tf_resource = aws_db_parameter_group(pg_identifier, **pg_values)
+            return aws_db_parameter_group(pg_identifier, **pg_values)
+
+        # 'deps' should contain a list of terraform resource names
+        # (not full objects) that must be created
+        # before the actual RDS instance should be created
+        deps = []
+
+        parameter_group = values.pop("parameter_group", None)
+        if parameter_group:
+            pg_tf_resource = populate_parameter_group(parameter_group)
             tf_resources.append(pg_tf_resource)
-            deps = self.get_dependencies([pg_tf_resource])
-            values["parameter_group_name"] = pg_name
+            deps += self.get_dependencies([pg_tf_resource])
+            # Associate parameter group to db instance
+            values["parameter_group_name"] = pg_tf_resource.get("name")
+
+        new_parameter_group = values.pop("new_parameter_group", None)
+        if new_parameter_group:
+            # discourage using new_parameter_group if parameter_group field is not utilized.
+            if parameter_group is None:
+                raise ValueError(
+                    "Cannot use new_parameter_group field without parameter_group."
+                    "This field is only used during RDS major version upgrade"
+                )
```

Comment:
this is a nice addition. you can implement also implement a dependency in the schemas such that `new_parameter_group` depends on `parameter_group`.

Reply:
That is a good idea! Let me look into this.

Reply:
I updated qontract-schemas to add dependency on the field. I'm keeping this check so this is compatible with older schema versions if we need to.